### PR TITLE
fix: [CDS-69754]: Do not call API for dropdown when dependent fields are not fixed

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger standards

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/GoogleArtifactRegistry/GoogleArtifactRegistry.tsx [hunk 1]
```diff
@@ -341,7 +342,6 @@ function FormComponent(
           <div className={css.imagePathContainer}>
             <FormInput.MultiTypeInput
               selectItems={getBuilds()}
-              disabled={!isAllFieldsAreFixed()}
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/GoogleArtifactRegistry/GoogleArtifactRegistry.tsx:344]: Do we need not disable this field?

Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/GoogleArtifactRegistry/GoogleArtifactRegistry.tsx:344]: no, as user can still make it runtime or provide manual values

**Comment by satyam-harness** [comment]
trigger rebase

**Review by pankajsingh-harness — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Pull request merged**